### PR TITLE
QtRvSim: init at 0.9.1

### PR DESCRIPTION
**Pull request by jdupak**
###### Motivation for this change

Packaging of a new application developed for educational purposes at the Czech Technical University to make it easily available to students, visitors of Embedded World Conference 2022 (where it will be presented) and other users.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS** test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] **New package, nothing depends**: Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes) **New package, none applicable.**
  - [] (Package updates) Added a release notes entry if the change is major or breaking
  - [] (Module updates) Added a release notes entry if the change is significant
  - [] (Module addition) Added a release notes entry if adding a new NixOS module
  - [] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jdupak** [comment]
@wentasah Could you help me with this package?

**Comment by wentasah** [comment]
AFAICT, it looks good. I'll be able to test it in the evening. But I'm not a commiter, so there is not much more I could do. Are you in time pressure or we have time? Nixpkgs comiters usually act within few days at most.

**Comment by jdupak** [comment]
> Are you in time pressure or we have time?

We have time.

**Comment by jdupak** [comment]
> AFAICT, it looks good. I'll be able to test it in the evening. 

Thanks

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14037,4 +14037,10 @@
     github = "nigelgbanks";
     githubId = 487373;
   };
+  jdupak = {
```
Comment (wentasah) [inline maintainers/maintainer-list.nix:14040]: According to the comment at the beginning of the file, this list of maintainers should be sorted alphabetically. I see that there are a few exceptions at the end of the file, but the rest is sorted.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -34728,4 +34728,6 @@ with pkgs;
   honeytail = callPackage ../servers/tracing/honeycomb/honeytail { };
 
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
+
+  qtrvsim = libsForQt5.callPackage ../applications/science/computer-architecture/qtrvsim {};
```
Comment (wentasah) [inline pkgs/top-level/all-packages.nix:34732]: `all-packages.nix` should be mostly sorted - see the comment at the top.

**Review by wentasah — CHANGES_REQUESTED** [review]
See the few in-line comments below.

Another minor thing (for the upstream project) is to specify full paths in the `.desktop` file. For the `Exec=` line it would be perhaps something like this:
```
Exec=@CMAKE_INSTALL_PREFIX@/bin/@MAIN_PROJECT_NAME_LOWER@_gui
```

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (jdupak) [reply pkgs/top-level/all-packages.nix]: ~fixed (the q letter area is not well sorted)

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (jdupak) [reply maintainers/maintainer-list.nix]: fixed

**Comment by wentasah** [comment]
I'm not sure whether there is some nixpkgs policy about `.desktop` files (cannot find any). [Freedesktop spec](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html) says that the `Exec` value may be either the absolute path or just a filename. And given that with Nix, its easy to work with multiple versions of the same package, it's beneficial to specify the paths as absolute, because otherwise, it might run some random version. On my NixOS, the following programs use an absolute path in the .desktop file: qtcreator, javaws, vlc, flameshot, kernelshark, cura.

The installation prefix is known at configure time. This is how Nix calls cmake:
```
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF
-DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF
-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release
-DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF
-DCMAKE_INSTALL_LOCALEDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/share/locale
-DCMAKE_INSTALL_LIBEXECDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/libexec
-DCMAKE_INSTALL_LIBDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/lib
-DCMAKE_INSTALL_DOCDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/share/doc/QtRVSim
-DCMAKE_INSTALL_INFODIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/share/info
-DCMAKE_INSTALL_MANDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/share/man
-DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/include
-DCMAKE_INSTALL_INCLUDEDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/include
-DCMAKE_INSTALL_SBINDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/sbin
-DCMAKE_INSTALL_BINDIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/bin
-DCMAKE_INSTALL_NAME_DIR=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1/lib
-DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT=
-DCMAKE_FIND_FRAMEWORK=LAST
-DCMAKE_STRIP=/nix/store/pja9g36cy32z3d51942jqk91a6l2d5nv-gcc-wrapper-10.3.0/bin/strip
-DCMAKE_RANLIB=/nix/store/sga0l55gm9nlwglk79lmihwb2bpv597j-binutils-2.35.2/bin/ranlib
-DCMAKE_AR=/nix/store/sga0l55gm9nlwglk79lmihwb2bpv597j-binutils-2.35.2/bin/ar
-DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++
-DCMAKE_INSTALL_PREFIX=/nix/store/s95zvxbi5vm06s7v19w3yfbdkzq4gzxf-QtRVSim-0.9.1
```
As you can see `CMAKE_INSTALL_PREFIX` is set at configure time and [it defaults to `/usr/local`](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html).

**Comment by wentasah** [comment]
But having an absolute path in .desktop is not required for inclusion into nixpkgs. It's only nice to have for people using multiple versions. And usually, those people can use the command line, so feel free to ignore that suggestion.

**Comment by wentasah** [comment]
Result of `nixpkgs-review pr 163128` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qtrvsim</li>
  </ul>
</details>

**Review by wentasah — APPROVED** [review]

**Comment by wentasah** [comment]
As I learned [elsewhere](https://github.com/NixOS/nixpkgs/pull/163581#discussion_r824097159), this package might also benefit from setting [`meta.mainProgram`](https://nixos.org/manual/nixpkgs/unstable/#var-meta-mainProgram).

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -5619,6 +5619,12 @@
     github = "jduan";
     githubId = 452450;
   };
+  jdupak = {
```
Comment (IvarWithoutBones) [inline maintainers/maintainer-list.nix:5622]: This should be in a seperate commit

**Review by IvarWithoutBones — COMMENTED** [review]
Just one suggestion, could you run `nixpkgs-fmt` on this as well? Just to make formatting a bit more consistent with the rest of nixpkgs. Would also love to see `meta.mainProgram` as @wentasah suggested :)

**Comment by IvarWithoutBones** [comment]
> But having an absolute path in .desktop is not required for inclusion into nixpkgs. It's only nice to have for people using multiple versions. And usually, those people can use the command line, so feel free to ignore that suggestion.

I learned recently that absolute paths actually breaks some stuff on Plasma, see https://github.com/NixOS/nixpkgs/pull/163468#event-6211981066. It does seem like a better solution to me if multiple versions are commonly used though as you said.

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (jdupak) [reply maintainers/maintainer-list.nix:5622]: Done

**Comment by jdupak** [comment]
> Just one suggestion, could you run `nixpkgs-fmt` on this as well? Just to make formatting a bit more consistent with the rest of nixpkgs. Would also love to see `meta.mainProgram` as @wentasah suggested :)

Reformated and added mainProgram.

File: pkgs/applications/science/computer-architecture/qtrvsim/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  version = "0.9.1";
+  src = fetchFromGitHub {
+    owner = "cvut";
+    repo = "qtrvsim";
+    rev = "refs/tags/v${version}";
+    sha256 = "AOksVS0drIBnK4RCxZw40yVxf4E8GjG9kU0rIZsY9gA=";
+  };
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+  buildInputs = [ qtbase ];
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/computer-architecture/qtrvsim/default.nix:13]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, wrapQtAppsHook, qtbase }:

stdenv.mkDerivation rec {
  pname = "QtRVSim";
  version = "0.9.1";

  src = fetchFromGitHub {
    owner = "cvut";
    repo = "qtrvsim";
    rev = "refs/tags/v${version}";
    sha256 = "AOksVS0drIBnK4RCxZw40yVxf4E8GjG9kU0rIZsY9gA=";
  };

  nativeBuildInputs = [ cmake wrapQtAppsHook ];

  buildInputs = [ qtbase ];

  meta = with lib; {
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**